apps/server/src/modules/fwu-learning-contents/controller/fwu-learning-contents.controller.ts

```diff
@@ -0,0 +1,31 @@
+import { Controller, Get, Req, Res, InternalServerErrorException, Param } from '@nestjs/common';
+import { ApiTags } from '@nestjs/swagger';
+import { Authenticate } from '@src/modules/authentication/decorator/auth.decorator';
+import { Request, Response } from 'express';
+import { ConfigService } from '@nestjs/config';
+import { GetFwuLearningContentParams } from './dto/fwu-learning-contents.params';
+import { FwuLearningContentsUc } from '../uc/fwu-learning-contents.uc';
+
+@ApiTags('fwu')
+@Authenticate('jwt')
+@Controller('fwu')
+export class FwuLearningContentsController {
+	constructor(
+		private readonly fwuLearningContentsUc: FwuLearningContentsUc,
+		private readonly configService: ConfigService
+	) {}
+
+	@Get('*/:fwuLearningContent')
+	async get(@Req() req: Request, @Res() res: Response, @Param() params: GetFwuLearningContentParams) {
+		if (!this.configService.get<boolean>('FEATURE_FWU_CONTENT_ENABLED')) {
+			throw new InternalServerErrorException('Feature FWU content is not enabled.');
+		}
+		const path = `${req.params[0]}/${params.fwuLearningContent}`;
+		const response = await this.fwuLearningContentsUc.get(path);
+		const startIndexOfContentType = params.fwuLearningContent.lastIndexOf('.');
+		const contentType = params.fwuLearningContent.slice(startIndexOfContentType);
+
+		res.type(contentType);
```

Comment:
this is defined by the client? can't you get this data from file/server side?

Reply:
The S3 client returns a "binary/octet-stream" as contentType for each file.

Reply:
then the upload is fucked up. 
when you upload a file you specify the content type, while downloading you kinda can receive that content type again, see schulcloud server file service code

Reply:
For this part are @Loki-Afro and @SevenWaysDP more expirenced then me. 
I only want to mentioned that it is only download for now, if it make any differents.

But the code to get the content should be moved to a private method. It can be stay in controller level but it is also not the best place.

Reply:
There is now a new function "getContentType" in the UC for the contentType.
This function returns the contentType of the response from the S3.